src/wireviz/wireviz.py

```diff
@@ -251,7 +253,7 @@ def main():
         file_out = args.output_file
     file_out = os.path.abspath(file_out)
 
-    parse(yaml_input, file_out=file_out)
+    parse(yaml_input, base_path=Path(args.input_file).parent, file_out=file_out)
```

Comment:
This `yaml_input` might be a concatenation of a prepended file and the main input file, and these two files might not be located in the same folder. Any image with relative path in the prepended YAML file might in that case not be found using the `base_path` from the main input file only. See use case in https://github.com/formatc1702/WireViz/issues/220#issuecomment-797154027 by @gmkado.

I suggest solving this in a way that might also support a set of prepended files.

Reply:
How about resolving paths in the prepended file and the main file separately, before concatenating? I am open for a new PR, based on this one, showing the implementation. Again, apologies for not tackling this myself.

Reply:
- I have not found a way to parse the files separately (with different base paths) and then join them into the same document, except maybe (load+resolve+dump) each file and then join the results before loading the combined dumps, but I have not tested something like this, and I don't know if some information might get lost in the process.
- A completely different approach is to put the different base paths into a path list used as the value of the [imagepath](https://graphviz.org/doc/info/attrs.html#d:imagepath) attribute, keep each image path relative, and trust Graphviz to resolve the paths. The path list should be in reverse order so the base path of the main file is first, because if the same image file name is available from more than one base path, the one in the base path of the main file should have priority. A side-effect is that this approach might remove the need for two separate `.gv` files (or at least make it easier to create the second from the first without generating everything twice).
- The same path list can of course be used to resolve each image path in WireViz, but then we loose the side-effect.

Reply:
See my [comment in the main discussion](https://github.com/formatc1702/WireViz/pull/189#issuecomment-932752516) and please mark resolved if you agree.

Reply:
The GraphViz `imagepath` approach sounds nice in theory, but when you start needing a different separator depending on the OS, it gets ugly fast.